overlord/hookstate/ctlcmd/get_interface_hooks_test.go

```diff
@@ -0,0 +1,90 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd_test
+
+import (
+	"github.com/snapcore/snapd/overlord/hookstate"
+	"github.com/snapcore/snapd/overlord/hookstate/ctlcmd"
+	"github.com/snapcore/snapd/overlord/hookstate/hooktest"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap"
+
+	"github.com/snapcore/snapd/interfaces"
+	. "gopkg.in/check.v1"
+)
+
+type getAttrSuite struct {
+	mockContext *hookstate.Context
+	mockHandler *hooktest.MockHandler
+}
+
+var _ = Suite(&getAttrSuite{})
+
+func (s *getAttrSuite) SetUpTest(c *C) {
+	s.mockHandler = hooktest.NewMockHandler()
+
+	state := state.New(nil)
+	state.Lock()
+
+	ch := state.NewChange("mychange", "mychange")
+
+	attrsTask := state.NewTask("connect-task", "my connect task")
+	attrsTask.Set("plug", &interfaces.PlugRef{Snap: "a", Name: "aplug"})
+	attrsTask.Set("slot", &interfaces.SlotRef{Snap: "b", Name: "bslot"})
+	attrs := make(map[string]interface{})
+	attrs["foo"] = "bar"
+	attrs["baz"] = []string{"a", "b"}
+	attrsTask.Set("plug-attrs", attrs)
+	attrsTask.Set("slot-attrs", make(map[string]interface{}))
+
+	hookTask := state.NewTask("run-hook", "my test task")
+	setup := &hookstate.HookSetup{Snap: "test-snap", Revision: snap.R(1), Hook: "prepare-plug-aplug"}
+	ch.AddTask(attrsTask)
+	state.Unlock()
+
+	var err error
+	s.mockContext, err = hookstate.NewContext(hookTask, setup, s.mockHandler)
+	c.Assert(err, IsNil)
+
+	s.mockContext.Lock()
+	s.mockContext.Set("attrs-task", attrsTask.ID())
+	defer s.mockContext.Unlock()
+	ch.AddTask(hookTask)
+}
+
+func (s *getAttrSuite) TestCommand(c *C) {
```

Comment:
Testing feels a bit lightweight, comparing the amount of logic in that function and the amount of logic tested in this file. I don't see --slot or --plug support, for example, and there's definitely much more missing.

Can we please get some more testing for aspects which are currently not covered?